go-controller/pkg/crd/types/networkselector.go

```diff
@@ -0,0 +1,88 @@
+package types
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NetworkSelectors selects multiple sets of networks.
+// +kubebuilder:validation:MinItems=1
+// +kubebuilder:validation:MaxItems=5
+// +patchMergeKey=networkSelectionType
+// +patchStrategy=merge
+// +listType=map
+// +listMapKey=networkSelectionType
+type NetworkSelectors []NetworkSelector
+
+// NetworkSelector selects a set of networks.
+// +kubebuilder:validation:XValidation:rule="!has(self.clusterUserDefinedNetworkSelector) || has(self.networkSelectionType) && self.networkSelectionType == 'ClusterUserDefinedNetworks'",message="Can only specify clusterUserDefinedNetworkSelector when networkSelectionType is ClusterUserDefinedNetworks"
```

Comment:
I think that without using an applyconfiguration it would be difficult to create the NetworkSelector because the Selectors will always be set to an empty value, `omitempty` doesn't work on structs.

Reply:
Can you expand. I don't really follow?

Reply:
When using a client-go that tries to set a `NetworkSelector` through the go structs it will always send all selectors because an empty label selector is not actually omitted, it results in `selector: {}` json representation.

Reply:
So you are saying that if we try to create something like this in go it will fail
```
ratypes.RouteAdvertisements{
		ObjectMeta: metav1.ObjectMeta{
			Name: tra.Name,
		},
		Spec: ratypes.RouteAdvertisementsSpec{
			TargetVRF:      tra.TargetVRF,
			Advertisements: []ratypes.AdvertisementType{ratypes.PodNetwork},
			NetworkSelectors: apitypes.NetworkSelectors{apitypes.NetworkSelector{NetworkSelectionType: apitypes.DefaultNetwork}}
		},
	}
```
because the other NetworkSelector fields are not really empty. I will try it out. 

There is an `omitzero` option, wonder if that would work. I don't see any other alternatives, do you?

Reply:
Exactly, something very similar already happens for RAs: https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5130#discussion_r2007116178

The only way out of this is to use pointers, this seems to be a requirement for discriminated unions so two birds with one stone?

Reply:
> Exactly, something very similar already happens for RAs: [#5130 (comment)](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5130#discussion_r2007116178)

I guess I am fixing this as well.